packages/gzip/fix-gzexe.patch

```diff
@@ -0,0 +1,171 @@
+diff -uNr gzip-1.9/gzexe.in gzip-1.9.mod/gzexe.in
+--- gzip-1.9/gzexe.in	2018-01-07 07:05:21.000000000 +0200
++++ gzip-1.9.mod/gzexe.in	2018-03-05 18:34:06.258486200 +0200
+@@ -66,7 +66,7 @@
+   esac
+ done
+ 
+-if test $# -eq 0; then
++if [ $# -eq 0 ]; then
+   printf >&2 '%s\n' "$0: missing operand
+ Try \`$0 --help' for more information."
+   exit 1
+@@ -74,7 +74,7 @@
+ 
+ tmp=
+ trap 'res=$?
+-  test -n "$tmp" && rm -f "$tmp"
++  [ -n "$tmp" ] && rm -f "$tmp"
+   (exit $res); exit $res
+ ' 0 1 2 3 5 10 13 15
+ 
+@@ -85,22 +85,22 @@
+   -*) file=./$i;;
+   *)  file=$i;;
+   esac
+-  if test ! -f "$file" || test ! -r "$file"; then
++  if [ ! -f "$file" ] || [ ! -r "$file" ]; then
+     res=$?
+     printf >&2 '%s\n' "$0: $i is not a readable regular file"
+     continue
+   fi
+-  if test $decomp -eq 0; then
++  if [ $decomp -eq 0 ]; then
+     if sed -e 1d -e 2q "$file" | grep "^skip=[0-9][0-9]*$" >/dev/null; then
+       printf >&2 '%s\n' "$0: $i is already gzexe'd"
+       continue
+     fi
+   fi
+-  if test -u "$file"; then
++  if [ -u "$file" ]; then
+     printf >&2 '%s\n' "$0: $i has setuid permission, unchanged"
+     continue
+   fi
+-  if test -g "$file"; then
++  if [ -g "$file" ]; then
+     printf >&2 '%s\n' "$0: $i has setgid permission, unchanged"
+     continue
+   fi
+@@ -112,10 +112,10 @@
+     printf >&2 '%s\n' "$0: $i might depend on itself"; continue;;
+   esac
+ 
+-  dir=`dirname "$file"` || dir=$TMPDIR
+-  test -d "$dir" && test -w "$dir" && test -x "$dir" || dir=/tmp
+-  test -n "$tmp" && rm -f "$tmp"
+-  if test -z "$mktemp_status"; then
++  dir=$(dirname "$file") || dir=$TMPDIR
++  [ -d "$dir" ] && [ -w "$dir" ] && [ -x "$dir" ] || dir=@TERMUX_PREFIX@/tmp
++  [ -n "$tmp" ] && rm -f "$tmp"
++  if [ -z "$mktemp_status" ]; then
+     type mktemp >/dev/null 2>&1
+     mktemp_status=$?
+   fi
+@@ -123,8 +123,8 @@
+     */) ;;
+     *) dir=$dir/;;
+   esac
+-  if test $mktemp_status -eq 0; then
+-    tmp=`mktemp "${dir}gzexeXXXXXXXXX"`
++  if [ $mktemp_status -eq 0 ]; then
++    tmp=$(mktemp "${dir}gzexeXXXXXXXXX")
+   else
+     tmp=${dir}gzexe$$
+   fi && { cp -p "$file" "$tmp" 2>/dev/null || cp "$file" "$tmp"; } || {
+@@ -132,7 +132,7 @@
+     printf >&2 '%s\n' "$0: cannot copy $file"
+     continue
+   }
+-  if test -w "$tmp"; then
++  if [ -w "$tmp" ]; then
+     writable=1
+   else
+     writable=0
+@@ -142,32 +142,35 @@
+       continue
+     }
+   fi
+-  if test $decomp -eq 0; then
++  if [ $decomp -eq 0 ]; then
+     (cat <<'EOF' &&
+-#!/bin/sh
+-skip=44
++#!@TERMUX_PREFIX@/bin/sh
++skip=52
++set -e
++
++export PATH="@TERMUX_PREFIX@/bin:@TERMUX_PREFIX@/bin/applets:${PATH}"
```

Comment:
Also is the backslash/test style change really *necessary*? I mean, it might be a good thing to do, but not in a downstream patch.

Reply:
> Also is the backslash/test style change really necessary?

I'll remove this.

Reply:
Btw by "backslash" I meant backquote. Sorry for the mistake.